### PR TITLE
DateTime: support 64bit unix epoch in DateTime.unix_time()

### LINKED ISSUE #658: Question: DateTime policy
すみません、`DateTime`について教えてください。

現在、unix_time()で返すタイムスタンプは2038 年までになっていますが、これは32bit unix epochの上限ですよね。

これに制限している理由は何なのでしょうか?

64bitとその制限が近づいてきたことを考えると、vim自体はまだとはいえ、この関数では「num64下である」などの条件付きで64bit unix epochの限界値まで対応してもよいのでないか、と思うのですが...

### PR DESCRIPTION
**Pull request by tsuyoshicho**
This PR is fix #658 

- check num64
- add test num64 and not num64
- document update

### CODE REVIEW

File: doc/vital/DateTime.txt [hunk 1]
```diff
@@ -159,6 +159,8 @@ DateTime.quarter()		*Vital.DateTime-DateTime.quarter()*
 DateTime.unix_time()		*Vital.DateTime-DateTime.unix_time()*
 	Returns the unix time.
 	If this DateTime is out of range of unix time, returns -1.
+	NOTE: When `num64` has supported. do not has "Year 2038 problem", work
```
Comment (ujihisa) [inline doc/vital/DateTime.txt]:
```suggestion
	The upper limit is 2038, but if your Vim has `+num64` it's limitless.
```

File: doc/vital/DateTime.txt [hunk 2]
```diff
@@ -159,6 +159,8 @@ DateTime.quarter()		*Vital.DateTime-DateTime.quarter()*
...
+	correctly.
```
Comment (ujihisa) [inline doc/vital/DateTime.txt]:
```suggestion
```

File: test/DateTime.vimspec [hunk 3]
```diff
@@ -355,6 +355,22 @@ Describe DateTime
         let new_dt = DT.from_unix_time(unix_time, dt.timezone())
         Assert Equals(new_dt.to_string(), dt.to_string())
       End
+
+      Context when DateTime is in the range of 64bit unix time
```
Comment (ujihisa) [inline test/DateTime.vimspec]:
```suggestion
      Context when DateTime is beyond 2038 but within the range of 64bit unix time
```

File: doc/vital/DateTime.txt (hunk 1 above)
Comment (tsuyoshicho) [reply doc/vital/DateTime.txt]: Thanks a lot. fix it.

File: doc/vital/DateTime.txt (hunk 2 above)
Comment (tsuyoshicho) [reply doc/vital/DateTime.txt]: Thanks a lot. fix it.

File: test/DateTime.vimspec (hunk 3 above)
Comment (tsuyoshicho) [reply test/DateTime.vimspec]: Thanks a lot. fix it.

**Pull request merged**